test/e2e/upgrade.go

```diff
@@ -61,6 +62,7 @@ func runSimpleUpgradeFlowWorkerNodeVersionForBareMetal(test *framework.ClusterE2
 }
 
 func runInPlaceUpgradeFlow(test *framework.ClusterE2ETest, clusterOpts ...framework.ClusterE2ETestOpt) {
+	test.T.Setenv(features.VSphereInPlaceEnvVar, "true")
```

Comment:
Why set the env variable in the flow? I would recommend setting it in each of the tests.